client/blocks/app-promo/style.scss

```diff
@@ -1,59 +1,73 @@
 .app-promo {
-	position: relative;
+	display: flex;
 	max-width: 300px;
-	margin: 0 auto 5px;
-	padding: 5px 0;
+	margin: 10px;
+	padding: 10px;
 	overflow: hidden;
+	background-color: var(--color-surface);
+	box-shadow: 0 0 0 1px rgba(var(--color-neutral-10-rgb), 0.5), 0 1px 2px var(--color-neutral-0);
+	border-radius: 2px;
 
 	@include breakpoint-deprecated( "<660px" ) {
 		display: none;
 	}
 }
 
-.app-promo .app-promo__icon {
-	position: absolute;
```

Comment:
I wanted to draw attention to some nuanced interactivity differences with the refactor away from absolute positioning. They are not necessarily incorrect, but there may be opportunity to improve the accessibility of the interactive elements for all users — cursors, keyboards, and screen readers. 

<details><summary>Interactive Element Size Changes</summary>

https://user-images.githubusercontent.com/438664/194336355-cc498d7c-7369-473e-a750-30e35563592e.mov

</details>

| Before | After |
| - | - |
| <img width="273" alt="inspection-before" src="https://user-images.githubusercontent.com/438664/194336504-4e825171-3681-42c1-b3fe-21ebbb7dabf5.png"> | <img width="264" alt="inspection-after" src="https://user-images.githubusercontent.com/438664/194336557-89e096f3-9e66-4789-a725-f7ac615c7df9.png"> |

As the above inspection images show, the sizes of the elements changed in the refactor of positioning. One possibility is that we could tweak the flex layout and padding so that the interactive elements more closely resemble previous sizes for cursor users and the focus state is more noticeable for keyboard users<sup>1</sup>. 

<details><summary>Interactive Element Size Proposal</summary>

https://user-images.githubusercontent.com/438664/194336419-2870810a-7068-424d-9379-e27d70817916.mov

</details>

<details><summary>Propose Code Changes Diff</summary>

```diff
diff --git a/client/blocks/app-promo/style.scss b/client/blocks/app-promo/style.scss
index ad90ec6ee8..d356581d77 100644
--- a/client/blocks/app-promo/style.scss
+++ b/client/blocks/app-promo/style.scss
@@ -1,8 +1,9 @@
 .app-promo {
+	align-items: flex-start;
 	display: flex;
 	max-width: 300px;
 	margin: 10px;
-	padding: 10px;
+	padding: 2px;
 	overflow: hidden;
 	background-color: var(--color-surface);
 	box-shadow: 0 0 0 1px rgba(var(--color-neutral-10-rgb), 0.5), 0 1px 2px var(--color-neutral-0);
@@ -15,8 +16,8 @@
 
 .app-promo .app-promo__dismiss {
 	cursor: pointer;
-	padding: 4px;
-	align-items: flex-start;
+	height: auto;
+	padding: 8px;
 
 	svg {
 		fill: var(--color-text-subtle);
@@ -28,6 +29,7 @@
 	height: auto;
 	border-radius: 0;
 	line-height: 1.5;
+	padding: 8px 0px 8px 8px;
 	color: var(--color-text);
 	font-size: $font-body-extra-small;
 	font-weight: 400;
```

</details>

---

1: The current production implementation does not have a noticeable focus state for the primary banner action and has the dismiss action as the first focused element. IMO improving the focus state is undeniable improve; The ordering of the primary and dismiss actions is debatable.

Reply:
Thank you for this David, I always appreciate the care you put into reviews and your focus on accessibility. I tested out your changes and agree that it feels like a good improvements. I've gone ahead to implement it in https://github.com/Automattic/wp-calypso/pull/68293/commits/679dfd0346009c47b928c3af94777979808caae4. 🙇‍♀️